### PR TITLE
Complete members only project.

### PR DESCRIPTION
**Pull request by RyelBanfield**
This PR contains the completed members-only project. The users have been built using Devise and users have the ability to signup, make a post, edit, delete. Once signed out you are not able to see which user made a post. You also cannot create a post without signing up.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+<br>
+
+<% if user_signed_in? %>
+  <%= link_to 'New Post', new_post_path %>
+  <%= link_to "Account", edit_user_registration_path %>
+  <%= link_to "Logout", destroy_user_session_path, method: :delete %>
+<% else %>
+  <%= link_to "Sign Up", new_user_registration_path %>
+  <%= link_to "Login", new_user_session_path %>
+<% end %>
```
Comment (MahmoudBakr23) [inline app/views/posts/index.html.erb:42]: Refactor the logic in this code into a rails helper, then recall it as ruby methods in here!

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+<% if current_user %>
+<p>
+  <strong>User:</strong>
+  <%= @post.user.username%>
+</p>
+<% end %>
+<% if current_user && current_user.username == @post.user.username %>
+  <td><%= link_to 'Edit', edit_post_path(@post) %></td>
+<% end %>
```
Comment (MahmoudBakr23) [inline app/views/posts/show.html.erb:14]: The same here!

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
# **Changes Required 🛑♻**

<hr>

Hello there :)
Great work and effort you put into this project. Well done 👍 
Your Linter is well-configured. 💯 
Descriptive README file and commit messages. 💯 
Controller's methods are set up correctly. 💯 
Good use of private keyword. 💯 
However, there are still a few more changes to make in order to get the approval. Keep up the great work ✔💪

<hr>

- [ ] Please make sure to include the signup and sign in buttons in all the screens (depending on whether or not the user is signed in, if not, make it log out and the username as link to their profile) Kindly write the proper logic as helper method then recall it in the application layout file.
sign_in_out_link_requirement

- [x] Kindly make sure to write another helper method to check if the user is signed_in?
- [x] Following the previous point, please move any logic code from the views to the helpers. There should be no logic in the views. See my comments below!
- [x] Kindly add validations to your models to prevent bad intention users. Currently, I can create an empty post, this shouldn't make any sense right? Please fix this issue.
- [ ] Thanks for making the extra work about showing, editing, and destroying a post. But, please try to stick to the requirements and make the routes' resources only for create, new, and index methods for posts, then refactor its controller's methods to reflect those changes in the routes.
posts_routes_requirement

<hr>

That's it, please do the required changes, then ask for another review.
Good luck and happy coding 👨‍💻💻

<hr>

Please feel free to contact me on Slack with the name "Mahmoud Bakr" if you had any questions regarding my review. 😊🙏

**Review by cyonii — APPROVED** [review]
# Status: Approved 👏

Congratulations 🎉 @RyelBanfield, great job implementing the changes requested by the previous reviewer, it takes great energy and effort.

Great project description with that humor 👏


## Recommended improvements (Optional)
Please note that these are optional suggestions in a bid to improve your project even further, this means you are not bound to implement them, but I recommend that you do.
- [x] User should be able to sign_out, your app breaks when I try to sign_out, please fix this behavior ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_out_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [x] The User model has\*many posts, in your case the user only has\*one posts *(not explicity said in specification - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_associations_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication) but with extra hint - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/associations_hint_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/) )\_

See you at the top 😎💡

> #### <br>You can ping me on Slack @[CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) to draw my attention in case I do not respond quickly on here. I'm always available and will be glad to help.<br><br>

**Comment by RyelBanfield** [comment]
I have fixed the User model however in regards to the ability to sign out, it works for me. - 
image

**Pull request merged**